packages/wagmi/src/utils/defaultWagmiCoreConfig.ts

```diff
@@ -33,46 +31,44 @@ export function defaultWagmiConfig({
   metadata,
   enableInjected,
   enableCoinbase,
-  enableEIP6963,
   enableEmail,
-  enableWalletConnect
-}: ConfigOptions) {
-  const { publicClient } = configureChains(chains, [
-    walletConnectProvider({ projectId }),
-    publicProvider()
-  ])
-
-  const connectors: Connector[] = []
+  enableWalletConnect,
+  enableEIP6963
+}: ConfigOptions): Config {
+  const connectors: CreateConnectorFn[] = []
 
   // Enabled by default
   if (enableWalletConnect !== false) {
-    connectors.push(
-      new WalletConnectConnector({ chains, options: { projectId, showQrModal: false, metadata } })
-    )
+    connectors.push(walletConnect({ projectId, metadata, showQrModal: false }))
   }
 
   if (enableInjected !== false) {
-    connectors.push(new InjectedConnector({ chains, options: { shimDisconnect: true } }))
-  }
-
-  if (enableEIP6963 !== false) {
-    connectors.push(new EIP6963Connector({ chains }))
+    connectors.push(injected({ shimDisconnect: true }))
   }
 
   if (enableCoinbase !== false) {
     connectors.push(
-      new CoinbaseWalletConnector({ chains, options: { appName: metadata?.name ?? 'Unknown' } })
+      coinbaseWallet({
+        appName: metadata?.name ?? 'Unknown',
+        appLogoUrl: metadata?.icons[0] ?? 'Unknown'
+      })
     )
   }
 
   // Dissabled by default
   if (enableEmail === true) {
-    connectors.push(new EmailConnector({ chains, options: { projectId } }))
+    connectors.push(emailConnector({ chains, options: { projectId } }))
   }
 
   return createConfig({
-    autoConnect: true,
+    chains: chains as [Chain, ...Chain[]],
+    client: ({ chain }) =>
+      createClient({
+        chain,
+        // TOD0: How to use WC transport? Do we need it for analytics?
```

Comment:
For this, what about wrapping the `http` function with a new function called `web3modalTransport`

the config could work like this:

```ts
export const config = createConfig({
  chains: [mainnet, sepolia],
  connectors: [injected()],
  transports: {
    [mainnet.id]: web3modalTransport(mainnet.id), 
    [sepolia.id]: web3modalTransport(sepolia.id), 
  },
})
```

internally

```ts
  client({ chain }) {
    return createClient({ chain, transport: web3modalTransport(chain.id) })
  },
```

Reply:
also, there are usually issues with our RPC because it doesn't support some chains.
Since we'll be getting the chain id here we could check and through a custom message like 
"Web3Modal Transport does not support chain X, please use a different Transport using the custom configuraiton"